core/src/serve_repair.rs

```diff
@@ -839,8 +856,10 @@ impl ServeRepair {
         nonce: Nonce,
     ) -> Result<Vec<u8>> {
         let sign_repairs_epoch = Self::sign_repair_requests_activated_epoch(root_bank);
-        let require_sig =
-            Self::should_sign_repair_request(request_slot, root_bank, sign_repairs_epoch);
+        let disable_sign_repairs_epoch =
+            Self::disable_sign_repair_requests_activated_epoch(root_bank);
+        let require_sig = Self::feature_active(request_slot, root_bank, sign_repairs_epoch)
```

Comment:
we perform this check twice.  might be good to keep the helper so we can reuse the logic